src/rust/metrics/src/lib.rs

```diff
@@ -2,188 +2,74 @@
 // Licensed under the Apache License, Version 2.0
 // http://www.apache.org/licenses/LICENSE-2.0
 
-pub use rustcommon_fastmetrics::*;
-use strum::IntoEnumIterator;
-use strum_macros::{AsRefStr, EnumIter};
+pub use rustcommon_metrics::{metric, Counter, Gauge};
 
-pub type Metrics = rustcommon_fastmetrics::Metrics<Stat>;
+#[doc(hidden)]
+pub extern crate rustcommon_metrics;
+#[doc(hidden)]
+pub use macros::to_lowercase;
 
-// As a temporary work-around for requiring all metrics to have a common type,
-// we combine server and storage metrics here.
+/// Create a set of metrics. Metrics are named with the lowercase name of the
+/// static that they are declared with.
+///
+/// # Example
+/// ```
+/// # use metrics::*;
+/// pelikan_metrics! {
+///     // Creates a counter metric called "my_metric" and constructs it by
+///     // calling Counter::new.
+///     static MY_METRIC: Counter;
+/// 
+///     // Creates a gauge metric called "some_other_metric" and initializes 
+///     // it to have the value 8.
+///     pub static SOME_OTHER_METRIC: Gauge = Gauge::with_value(8);
+/// }
+/// ```
+#[macro_export]
+macro_rules! pelikan_metrics {
```

Comment:
I'm again raising the very important (bike-shedding) question on naming 🤪, which is if we want to call the macro `static_metrics` instead of `pelikan_metrics`, given most modules in this project don't explicitly use `pelikan` in their name but generally stick to a functional description.

Reply:
I think `pelikan_metrics` is somewhat clearer but this isn't that important so I'll just rename it